### DISCUSSION TITLE
checkresiduals() vs Residuals Assumption Tests

### DISCUSSION THREAD

**Discussion by RanggaGemilang** [teamalgoritma/community#334] [Q&A] [answered]
Hi, saya menemukan perbedaan hasil uji asumsi residuals dari 2 metode yang saya gunakan.

1. Metode pengecekan dengan checkresiduals()

![checkresiduals](https://user-images.githubusercontent.com/107049922/191831294-2b2a6d82-4611-4499-bdff-1bda4c136cd7.png)

Berdasarkan hasil diatas, p-value yang dihasilkan oleh Box-Ljung adalah sebesar 0.007654 (indikasi autokorelasi) dan distribusinya terlihat normal.

2. Metode pengecekan dengan box.test() dan shapiro.test()

![residualjung](https://user-images.githubusercontent.com/107049922/191832012-5e5dff5a-4e8f-405c-84d2-308c707444fa.png)

![residualsapiro](https://user-images.githubusercontent.com/107049922/191832040-4bb619a7-3ca5-48ee-8c1c-5e933a27b1d0.png)

Berdasarkan hasil diatas, p-value menunjukkan hal yang sebaliknya, dimana p-value Box-Ljung berada diangka 0.9804 (tidak ada indikasi autokorelasi) dan p-value Shapiro diangka 0.000001169 (distribusi tidak normal)

Meskipun kedua metode diatas sama-sama tidak lolos uji asumsi residuals secara keseluruhan, apakah ada penjelasan yang bisa dijelaskan mengenai perbedaan hasil tersebut?

Terima kasih sebelumnya :)

**Comment by NabiilahArdini** [comment, marked as answer]
Halo @RanggaGemilang, hal tersebut seharusnya berkaitan dengan operasi matematika/parameter tertentu yang ada pada fungsinya. Pertanyaan Bapak sepertinya mirip dengan diskusi ini: https://stackoverflow.com/questions/49141746/different-results-from-forecastcheckresiduals-and-box-test, dari sana, ada parameter tambahan yang di perhitungkan pada fungsi `check_residuals()`. Semoga sharing ini dapat membantu. 

Apabila comment ini menjawab pertanyaan Bapak, silahkan klik tombol "Mark as Answer" pada comment ini. Terima kasih!
